File: src/leetCodeA/算法入门/第2天_双指针/_977_有序数组的平方.java
package leetCodeA.算法入门.第2天_双指针;

import java.util.Arrays;

public class _977_有序数组的平方 {

    public int[] sortedSquares(int[] nums) {
        /*for(int i = 0; i < nums.length; i++) {
            nums[i] = nums[i] * nums[i];
        }
        Arrays.sort(nums);
        return nums;*/
        int len = nums.length;
        int left = 0;
        int right = len - 1;
        int[] arr = new int[len];
        int index = right;
        int lValue;
        int RValue;
        while (left <= right) {
            lValue = nums[left] * nums[left];
            RValue = nums[right] * nums[right];
            if (lValue > RValue) {
                arr[index] = lValue;
                left++;
            } else {
                arr[index] = RValue;
                right--;
            }
            index--;
        }
        return arr;

    }
}

File: src/main/C/array/_977_有序数组的平方.c
int* sortedSquares(int* nums, int numsSize, int* returnSize){
    int left = 0;
    int right = numsSize - 1;
    int* arr = (int*)malloc(numsSize * sizeof(int));
    int index = right;
    *returnSize = numsSize;
    while (left <= right) {
        int valL = nums[left] * nums[left];
        int valR = nums[right] * nums[right];
        if (valL > valR) {
            arr[index] = valL;
            ++left;
        } else {
            arr[index] = valR;
            --right;
        }
        --index;
    }
    return arr;
}